src/util/server.c

```diff
@@ -264,6 +264,11 @@ static void default_quit(struct tevent_context *ev,
 {
     struct main_context *ctx = talloc_get_type(private_data, struct main_context);
     talloc_free(ctx);
+
+    /* It is ok to free ev (ctx->event_ctx; a top level context) here as well
+     * since we never go back to the event loop from here. */
+    talloc_free(ev);
+
     orderly_shutdown(0);
```

Comment:
This might change the order of destructors execution.

Currently it's `orderly_shutdown()` -> exit() -> at-exit-handlers -> talloc mem deallocation -> d-tors.

I don't say this change is wrong. But there are might be very subtle side effects (we had a crash in LDAP connection d-tor in this execution chain that @thalman fixed some time ago...)

Reply:
Probably this is a reason of integrations tests failing...